subprojects/packagefiles/openssl/generator.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+import glob
+import os
+import shutil
+import subprocess
+
+# Change to the directory of the Bash script
+script_dir = os.path.dirname(os.path.abspath(__file__))
+os.chdir(script_dir)
+
+# Node.js version should bundle OpenSSL of matching version to one specified in wrap file
+node_version = "v19.7.0"
+openssl_version = os.environ.get("OPENSSL_VERSION")
+
+if openssl_version is None:
+    with open("../../openssl.wrap", encoding="utf-8") as wrap_file:
+        for line in wrap_file:
+            if "directory = " in line:
+                openssl_version = line.strip().split()[-1]
+                break
+
+# Clone the Node.js repository
+subprocess.run(["git", "clone", "--depth", "1", "-b", node_version, "https://github.com/nodejs/node.git"])
```

Comment:
```suggestion
git = find_program('git')
subprocess.run([git, "clone", "--depth", "1", "-b", node_version, "https://github.com/nodejs/node.git"])
```

Reply:
wait a minute, this is horrible advice.

python has no find_program.